lib/internal/Magento/Framework/File/Uploader.php

```diff
@@ -604,10 +604,12 @@ private function _createDestinationFolder($destinationFolder)
     public static function getNewFileName($destinationFile)
     {
         $fileInfo = pathinfo($destinationFile);
-        if (file_exists($destinationFile)) {
+        $destinationBaseFile = str_replace('/tmp/','/',$destinationFile);
+        if (file_exists($destinationFile) || file_exists($destinationBaseFile)) {
             $index = 1;
             $baseName = $fileInfo['filename'] . '.' . $fileInfo['extension'];
-            while (file_exists($fileInfo['dirname'] . '/' . $baseName)) {
+            $dirBasename = str_replace('/tmp/','/',$fileInfo['dirname']);
+            while (file_exists($fileInfo['dirname'] . '/' . $baseName) || file_exists($dirBasename . '/' . $baseName)) {
```

Comment:
Why previous logic does not find a proper unique file name?

Reply:
each time we have to check file name in both destination folders like for category image we have check following destination.
pub/media/catalog/category
pub/media/catalog/tmp/category
in default it check only pub/media/catalog/tmp/category but if same file exist in pub/media/catalog/category so it replace it.

Reply:
Hello @orlangur

https://github.com/magento/magento2/pull/20763#issuecomment-469679520
It also check for above situation